src/query/processor.rs

```diff
@@ -122,14 +160,142 @@ impl<T: Transport + Clone> Processor<T> {
         let network = Network::new(self.transport.clone(), req.query_id, req.roles.clone());
         let gateway = Gateway::new(my_role, network, GatewayConfig::default()).await;
 
-        handle.set_state(QueryState::AwaitingInputs(gateway))?;
+        handle.set_state(QueryState::AwaitingInputs(req.config, gateway))?;
 
         Ok(())
     }
 
+    /// Receive inputs for the specified query. That triggers query processing
+    pub fn receive_inputs(&self, input: QueryInput) -> Result<(), QueryInputError> {
+        let mut queries = self.queries.inner.lock().unwrap();
+        match queries.entry(input.query_id) {
+            Entry::Occupied(entry) => {
+                let state = entry.remove();
+                match state {
```

Comment:
```suggestion
                match entry.remove() {
```

Reply:
there is actually a different issue - state needs to be inserted back if there is an error. so I would need `state` variable, but need to change `_` branch